functions/go/set-labels/transformer/labels.go

```diff
@@ -0,0 +1,164 @@
+	if err := transformer.Config(rl.FunctionConfig); err != nil {
+		rl.Results = append(rl.Results, fn.ErrorResult(err))
+		return false, nil
+	}
+	if err := transformer.Transform(rl.Items); err != nil {
+		rl.Results = append(rl.Results, fn.ErrorResult(err))
+		return false, nil
+	}
+
+	rl.Results = append(rl.Results, transformer.Results...)
+	return true, nil
+}
+
+// Config parse the functionConfig kubeObject to the fields in the LabelTransformer
+func (p *LabelTransformer) Config(functionConfig *fn.KubeObject) error {
+	// parse labels to NewLabels
+	switch {
+	case functionConfig.IsEmpty():
+		return fmt.Errorf("failed to configure function: `functionConfig` must be either a `ConfigMap` or `SetLabels`")
+	case functionConfig.IsGVK("", "v1", "ConfigMap"):
+		p.NewLabels = functionConfig.NestedStringMapOrDie("data")
+	case functionConfig.IsGVK(fnConfigGroup, fnConfigAPIVersion, fnConfigKind):
+		if _, exist, err := functionConfig.NestedSlice(fnDeprecateField); exist || err != nil {
+			return fmt.Errorf("`additionalLabelFields` has been deprecated")
+		}
+		p.NewLabels = functionConfig.NestedStringMapOrDie("labels")
+		if len(p.NewLabels) == 0 {
+			return fmt.Errorf("failed to configure function: input label list cannot be empty, required valid `labels` field")
+		}
+	default:
+		return fmt.Errorf("unknown functionConfig Kind=%v ApiVersion=%v, expect `%v` or `ConfigMap`",
+			functionConfig.GetKind(), functionConfig.GetAPIVersion(), fnConfigKind)
+	}
+	return nil
+}
+
+// set labels according to the generated common label
+func (p *LabelTransformer) setLabelsInSpecs(o *fn.KubeObject) error {
+	for _, spec := range CommonSpecs {
+		if o.IsGVK(spec.Identifier.group, spec.Identifier.version, spec.Identifier.kind) {
+			err := updateLabels(&o.SubObject, spec.Path, p.NewLabels, spec.CreateIfNotPresent)
+			p.LogResult(o, spec.Path)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// Transform updates the labels in the right path using configured logic
+func (p *LabelTransformer) Transform(objects fn.KubeObjects) error {
+	for _, o := range objects {
+		// this label need to set for all GKV
+		defaultPath := FieldPath{"metadata", "labels"}
+		err := updateLabels(&o.SubObject, defaultPath, p.NewLabels, true)
+		p.LogResult(o, defaultPath)
+		if err != nil {
+			return err
+		}
+		// set other common labels according to specific GKV
+		err = p.setLabelsInSpecs(o)
+		if err != nil {
+			return err
+		}
+		// handle special case with slice
+		if o.IsGVK("apps", "", "StatefulSet") {
+			for _, vctObj := range o.GetMap("spec").GetSlice("volumeClaimTemplates") {
+				err = updateLabels(vctObj, defaultPath, p.NewLabels, true)
+				p.LogResult(o, defaultPath)
+				if err != nil {
+					return err
+				}
+			}
+		}
+	}
+	return nil
+}
+
+// Logs the result of each operation, can also modify into other logs user wants
+func (p *LabelTransformer) LogResult(o *fn.KubeObject, path []string) {
+	res, _ := json.Marshal(p.NewLabels)
```

Comment:
The results generated this way is not always useful. It should only show the labels that have been added or updated.
For example, running the set-labels functions with the same config again should not generate any results since labels have been updated in the first run.
cc: @yuwenma